### PR TITLE
(feat) US-231: Support transfers

### PR DESCRIPTION
**Pull request by rrw-zilliqa**
(fix) US-231: tests now have a hardhat runtime environment, and there
      is a better way to select your network for testing.

### CODE REVIEW

File: test/zilliqa.live.test.ts [hunk 1]
```diff
@@ -1,35 +1,62 @@
...
   describe("Zilliqa network APIs", function () {
-    // I happen to know that this account exists - rrw 2023-03-12
-    it("Should be able to fetch a balance", async function () {
-      const account = zobj.getAccounts()[0];
-      const [bal, nonce] = await zobj.getBalance(account);
+    // Try an account that (hopefully) doesn't exist.
+    it("Should be able to fetch a balance from a nonexistent account", async function () {
+      const privKey = this.zobj.createPrivateKey();
+      const [ acc, idx ] = this.zobj.pushPrivateKey(privKey);
+      //console.log(`idx ${idx}`);
```
Comment (its-saeeed) [inline test/zilliqa.live.test.ts:20]: Can we remote this commented line?

Comment (its-saeeed) [reply test/zilliqa.live.test.ts:20]: and all commented ones? :)

File: src/type-extensions.ts [hunk 2]
```diff
@@ -29,5 +29,9 @@ declare module "hardhat/types/runtime" {
       init: Init
     ) => Promise<[Transaction, ScillaContract]>;
     zilliqa: ZilliqaHardhatObject;
+    getZilliqaChainId: () => number;
+    getNetworkUrl: () => string;
+    getPrivateKeys: () => string[];
+    setAccount: (num: number) => void;
```
Comment (its-saeeed) [inline src/type-extensions.ts:35]: Does it set the active account? Maybe `setActiveAccount` or `setDefaultSigner` is a better name.

File: src/ZilliqaHardhatObject.ts [hunk 3]
```diff
@@ -28,14 +28,54 @@ export class ZilliqaHardhatObject {
...
+    const tx1 = zjs.transactions.new(callParams);
+    const tx = zjs.blockchain.createTransaction(tx1, setup!.attempts, setup!.timeout)
+    return tx
+  }
+
+  public async getBalance(account: Account): Promise<[BN, Number]> {
+    return this.getBalanceForAddress(account.address);
+  }
+
+  public async getBalanceForAddress(a: string): Promise<[BN, Number]> {
```
Comment (its-saeeed) [inline src/ZilliqaHardhatObject.ts:76]: Please rename `a` to a more meaningful name.

**Review by its-saeeed — APPROVED** [review]

File: test/zilliqa.live.test.ts (hunk 1 above)
Comment (rrw-zilliqa) [reply test/zilliqa.live.test.ts:20]: OK, but they're in a test - it's not as though they're a big maintenance burden, and rewriting them all when something fails and you need to debug stuff is really tedious....

File: src/type-extensions.ts (hunk 2 above)
Comment (rrw-zilliqa) [reply src/type-extensions.ts:35]: Named after the function in `ScillaContractDeployer` that does this - perhaps we should rename that too? (there again, it is part of the API now). Renamed, anyway.

File: src/ZilliqaHardhatObject.ts (hunk 3 above)
Comment (rrw-zilliqa) [reply src/ZilliqaHardhatObject.ts:76]: Good point..

**Comment by its-saeeed** [comment]
@rrw-zilliqa LGTM!

**Pull request merged**